src/main/java/org/fenixedu/academic/domain/accounting/events/insurance/InsuranceEvent.java

```diff
@@ -74,6 +78,14 @@ private InsuranceEvent() {
     public InsuranceEvent(Person person, ExecutionYear executionYear) {
         this();
         init(EventType.INSURANCE, person, executionYear);
+        setupDueDate();
+    }
+
+    private void setupDueDate() {
+        final PostingRule postingRule = getPostingRule();
+        if (postingRule instanceof InsurancePR) {
```

Comment:
Should we log "anomaly" situations where the PostingRule for InsuranceEvents isn't an InsurancePR to avoid surprises later on?

Reply:
that is a good idea btw.